### ISSUE TITLE
I am an MS partner and have an EV Code signing cert

### ISSUE THREAD

**Issue by JMPZ11** [bobranten/Ext4Fsd#20]
I dropped the cheese on an EV code signing cert for my ill-fated start-up and haven't had a good opportunity to use it. This kind of project is exactly what I had in mind when I acquired it.

If you don't mind the drivers being signed by Stand Back Labs LLC, I'd love to help out. I will host them in a repo under my org @standbacklabs. Let me know if you would be interested in looping me into your releases!

**Comment by bobranten** [comment]
That would be great! Have you signed drivers before so you know the 
process?

Bo Branten

**Comment by JasonJShuler** [comment]
Nope. Virgin territory. I'm still trying to get it to build - turns out VS 2022 is kinda limited...

**Comment by JasonJShuler** [comment]
I am theoretically signed up as a hardware developer, so in theory I could submit it for WHQL or whatever

**Comment by JasonJShuler** [comment]
Ok making progress - seem to have the right requirements, but VS2019 has many errors when I try to build:
![image](https://user-images.githubusercontent.com/29778397/159589320-09d5093c-608c-4c91-9151-c5262ef1d4c8.png)

**Comment by JasonJShuler** [comment]
Ok, disabled treat warnings as errors and now it can't find #include "mountmgr.h" in enumDisk.cpp.

![image](https://user-images.githubusercontent.com/29778397/159593207-df71b80e-6ea5-45fd-a891-4c4da372e0df.png)

**Comment by JasonJShuler** [comment]
FYI - I'm stuck on this last one

**Comment by JasonJShuler** [comment]
ok, I got it to build and have the output files. Sounds like for Windows 10 and 11, I only need to do attestation signing whihc is actually pretty easy, but not what I expected. I pack into a cab, sign the cab, upload to partner center and supposedly 10 minutes later I can download a package now signed by Microsoft that will play nice with secure boot.

Any other signing methods apply to pre-Win 10. So I'm too tired to figure out the cab packing, but I am somewhat familiar with signtool - at least in the java world. Regroup tomorrow?

**Comment by bobranten** [comment]
That was nice progress. Thank you for youre time!

Bo Brantén

**Comment by bobranten** [comment]
Have you done any progress?

Bo Brantén

**Comment by kristibektashi** [comment]
@JasonJShuler How is signing the driver going? Because I need to use it on a system with Secure Boot enabled and would rather not enable testsigning

**Comment by bobranten** [comment]
> @bobranten @JasonJShuler any progress?

I have not heard from him in a while but it did sound promising to begin 
with...

Bo Branten

**Comment by bobranten** [comment]
> Ok making progress - seem to have the right requirements, but VS2019 has many errors when I try to build:
> ![image](https://user-images.githubusercontent.com/29778397/159589320-09d5093c-608c-4c91-9151-c5262ef1d4c8.png)

Those warnings is because I have only tryed to compile the driver with 
Win10 WDK but you used the latest Win11 WDK and some functions are marked 
deprecated, but that is only advisory warnings anyway and you managed to 
turn them off.

Bo Branten

**Comment by kristibektashi** [comment]
@JasonJShuler @bobranten Any progress? Is this project dead?

**Comment by bobranten** [comment]
I have not been able to reach the person who offered to sign the driver so 
there is no progres on that but the project itself is alive, I maintain 
the driver and try to fix bugs but signing it would be very helpful for 
the users.

Bo Branten

**Comment by kristibektashi** [comment]
> I have not been able to reach the person who offered to sign the driver so there is no progres on that but the project itself is alive, I maintain the driver and try to fix bugs but signing it would be very helpful for the users. Bo Branten

Maybe try emailing them? Based on their GitHub contributions they seem somewhat active

**Comment by sskras** [comment]
ping @JasonJShuler (just in case:)

**Comment by Shadowphoenix** [comment]
> > I have not been able to reach the person who offered to sign the driver so there is no progres on that but the project itself is alive, I maintain the driver and try to fix bugs but signing it would be very helpful for the users. Bo Branten
> 
> Maybe try emailing them? Based on their GitHub contributions they seem somewhat active

i checked the Website linked in the StandbackLabsrepo he linked above and it lead me to an entirely diffrenty site marked as fishy by ublockorigin which seemingly had nothing to do with what StandbackLabs claims to be based on their GitHub-Repo.

Not to be the boooo-man but this raises a few red flags for me.

@bobranten pinging for attention ;)

**Comment by kristibektashi** [comment]
@JasonJShuler @bobranten any progress in signing the driver?

**Comment by JasonJShuler** [comment]
Sorry all - my cert expired, and since Microsoft messed up my domain renewal, it was swiped by a scam squatter as noticed by @Shadowphoenix. The new domain is standbacklabs.org. To use a new domain, I will probably have to go through the vetting process again. I seem to still be setup as a MS hardware developer - don't know if I need an active CSS. I'll have to look.

Hopefully the squatter will not renew the domain in a few months, and I can get it back...

As I mentioned previously, I have zero experience with driver development - I will need guidance on how to properly build and test for any supported platforms.

Thanks!

**Comment by bobranten** [comment]
Sorry to hear about the problems but it is very nice of you to want to 
help us, we will see how it goes.

Bo Branten

**Comment by JasonJShuler** [comment]
FYI - looks like I should be able to renew the csc. It's a $300 investment and is only good for a year.., so... I hope someone will have some time to help me with the steps to get it certified.

**Comment by Zero3K** [comment]
I wonder if you could also get ERAM (A RAM Disk driver) to be signed so it can be used in Windows 10 and 11. Its source code is at https://github.com/Zero3K/eram.

**Comment by JasonJShuler** [comment]
Possibly - I haven't started the vetting yet - the .com domain registration
expires next month and I'm hoping the squatter doesn't renew (since my
previous cert was tied to the .com domain). Sorry about the weird stuff...

**Comment by netomx** [comment]
So, 300 a year?


Ernesto Villarreal

**Comment by JasonJShuler** [comment]
Yes, and I renewed it specifically for this... I am full of bright ideas 😜

**Comment by netomx** [comment]
So, if you get the cert, maybe you can accept donations?

I would like to have the app working.

Ernesto Villarreal

**Comment by bobranten** [comment]
It is great that you want to help us! Thank you.

Bo Branten

**Comment by kristibektashi** [comment]
@JasonJShuler It's been a few months. Have you gotten your domain back?

**Comment by sskras** [comment]
Ping @JasonJShuler :)

**Comment by Shadowphoenix** [comment]
if i read the graph correctly he has not been active in several months. i doubt that a ping would change that

**Comment by kristibektashi** [comment]
Ping again @JasonJShuler

**Comment by bobranten** [comment]
I have know found another person that will help me sign the driver, however I think this first try only loads on Windows 11, but please test it and report to me if it runs without enabling test mode! Install program https://www.accum.se/~bosse/ext2fsd/0.70/Ext2Fsd-setup-signed-win11.exe or only driver file to upgrade https://www.accum.se/~bosse/ext2fsd/0.70/ext2fsd.sys

**Comment by sskras** [comment]
Fails on my **w10 Pro 21H2**:
```
C:\Users\saukrs> sudo sc start ext2fsd
[SC] StartService FAILED 127:

The specified procedure could not be found.


C:\Users\saukrs> sc query ext2fsd

SERVICE_NAME: ext2fsd
        TYPE               : 2  FILE_SYSTEM_DRIVER
        STATE              : 1  STOPPED
        WIN32_EXIT_CODE    : 127  (0x7f)
        SERVICE_EXIT_CODE  : 0  (0x0)
        CHECKPOINT         : 0x0
        WAIT_HINT          : 0x0

C:\Users\saukrs> ver

Microsoft Windows [Version 10.0.19044.3086]

C:\Users\saukrs>
```

**Comment by bobranten** [comment]
Yes, I get the same error on Win10 but I think it is compiled for Win11 so 
we will hear if anyone test it there and then I can try to get a Win10 
version signed later.

Bo Branten

**Comment by netomx** [comment]
is it compiled for arm64 too? I am on one now.


Ernesto Villarreal

**Comment by zamtmn** [comment]
install on win11pro, worked
```
Y:\>sc query ext2fsd

Имя_службы: ext2fsd
        Тип                : 2  FILE_SYSTEM_DRIVER
        Состояние          : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, IGNORES_SHUTDOWN)
        Код_выхода_Win32   : 0  (0x0)
        Код_выхода_службы  : 0  (0x0)
        Контрольная_точка  : 0x0
        Ожидание           : 0x0

Y:\>ver

Microsoft Windows [Version 10.0.22631.3155]
```

**Comment by bobranten** [comment]
On Tue, 20 Feb 2024, Andrey wrote:

> install on win11pro, worked
> ```
> Y:\>sc query ext2fsd
>
> Имя_службы: ext2fsd
>        Тип                : 2  FILE_SYSTEM_DRIVER
>        Состояние          : 4  RUNNING


Greate! Then we have a signed version for Windows 11 at least.

Bo Branten

**Comment by netomx** [comment]
It seems it works on Win11 Arm64

C:\Windows\System32>sc query Ext2Srv

NOMBRE_SERVICIO: Ext2Srv
        TIPO               : 110  WIN32_OWN_PROCESS  (interactive)
        ESTADO             : 4  RUNNING
                                (STOPPABLE, NOT_PAUSABLE, ACCEPTS_SHUTDOWN)
        CÓD_SALIDA_WIN32   : 0  (0x0)
        CÓD_SALIDA_SERVICIO: 0  (0x0)
        PUNTO_COMPROB.     : 0x0
        INDICACIÓN_INICIO  : 0x0

C:\Windows\System32>


Ernesto Villarreal

**Comment by netomx** [comment]
Cant mount on ARM64.


Ernesto Villarreal

**Comment by bobranten** [comment]
On Tue, 20 Feb 2024, netomx wrote:

> Cant mount on ARM64.

I think it was only the help program that started as an emulated program, 
the driver needs to be recompiled for ARM64, I will look into that.

Bo Branten

**Comment by netomx** [comment]
Confirmed it installs and service is active on Win11 Pro x64.Will test
later if I can mount it.


Ernesto Villarreal

**Comment by zamtmn** [comment]
everything works fine for me, partitions are mounted

**Comment by bobranten** [comment]
Now we have managed to make a signed install program for Windows 10: https://www.accum.se/~bosse/ext2fsd/0.70/Ext2Fsd-setup-signed-win10.exe
For they who do manual install signed driver files are here (even ARM/ARM64) https://www.accum.se/~bosse/ext2fsd/0.70/signed/
I will try to make an install program for ARM/ARM64 later.
/Bo Branten

**Comment by mirh** [comment]
Smooth as silk on my W10 too.

**Comment by OLuserGi** [comment]
Hi!
I use your wonderful DRIVER every day!!! Thank you very much!
I am using the boot mode of the system driver. To do this, press the F8 key every time you boot. The signed one can solve the problem and turning on the computer will not require any clicks from me.

The new signed version contains a valid certificate. I imported this certificate into trusted publishers, but got an error during the reinstallation. The error says that Microsoft does not consider the publisher to be trusted.

**Comment by OLuserGi** [comment]
> Hi! I use your wonderful DRIVER every day!!! Thank you very much! I am using the boot mode of the system driver. To do this, press the F8 key every time you boot. The signed one can solve the problem and turning on the computer will not require any clicks from me.
> 
> The new signed version contains a valid certificate. I imported this certificate into trusted publishers, but got an error during the reinstallation. The error says that Microsoft does not consider the publisher to be trusted.

After reboot my Win11 not question to press F8. All good!

But after login I take window with question about give rights start ext2srv.exe, becouse it not trusted. What I can do with this?
